charmcraft/helptexts.py

```diff
@@ -118,3 +133,116 @@ def get_full_help(command_groups, global_options):
     # join all stripped blocks, leaving ONE empty blank line between
     text = '\n\n'.join(block.strip() for block in textblocks) + '\n'
     return text
+
+
+def get_detailed_help(command_groups, global_options):
+    """Produce the text for the detailed help.
+
+    It has the following structure:
+
+    - usage
+    - summary
+    - global options
+    - all commands shown with description, grouped
+    - more help
```

Comment:
Same comment as below. It would be nice to have the expected type/structure of function parameters described, since it makes the code easier to understand with them in mind, even though the structure can be inferred from the code and tests.

Reply:
Thanks, I improved the docstring with the suggestion.